deployment/debian-x64/package.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+
+source ../build.common.sh
+
+VERSION=`get_version ../..`
+
+package_portable ../.. `pwd`/dist/jellyfin_${VERSION}
+
+# Build a Jellyfin .deb file with Docker on Linux
+# Places the output .deb file in the parent directory
+
+package_temporary_dir="`pwd`/pkg-dist-tmp"
+output_dir="`pwd`/pkg-dist"
+current_user="`whoami`"
+image_name="jellyfin-debuild"
+
+cleanup() {
+    set +o errexit
+    docker image rm $image_name --force
+    rm -rf "$package_temporary_dir"
+}
+trap cleanup EXIT INT
+
+docker build ../.. -t "$image_name" -f ./Dockerfile --build-arg SOURCEDIR="/jellyfin-${VERSION}"
+mkdir -p "$package_temporary_dir"
+mkdir -p "$output_dir"
+docker run --rm -v "$package_temporary_dir:/temp" "$image_name" sh -c 'find / -maxdepth 1 -type f -name "jellyfin*" -exec mv {} /temp \;'
+chown -R "$current_user" "$package_temporary_dir"
```

Comment:
Should you not instead add a bit at the beginning requesting root if it doesn't have it already? I personally don't like the idea of sudo's in a script. I might have used it recently and now its going to run a script with my cached creds, potentially doing something I don't want.

I have a script I wrote for another project that handled this at least in part. Can see it here: https://github.com/morph1904/TygerCaddy/blob/master/tyger-install-ubuntu.sh

Reply:
It should all write to user writable folders. So sudo should not be required. Since both $output_dir and $package_temporary_dir should be within the source tree. (Which you should have write access to.

Sudo in a script is a bad idea IMO.

Reply:
`docker` itself requires root access for *all* commands. You can add yourself to the docker group, but thats basically the same as granting yourself root access to the entire system since you can mount system files as volumes and edit them in a container.

It's best to assume that admins and security conscious folks will *not* be adding themselves to the docker group for the above reason.

Reply:
On any of the default installs of docker on dev machines, this works just like it is. Both Docker for Windows (with the client in WSL) and Docker for Mac, handle this out of the box. I suppose one could run the whole script as root.

Reply:
Ah, well... I know at least @joshuaboniface and I use Linux as our dev machine and by default, Linux requires root access.

Reply:
Itt needs to be sudo'd because the file moved from the `docker` container ends up owned `root:root`, so the unprivileged user running the command has no perms to edit (and hence `mv`) it. Or chown has to happen in docker but that might not work because the user doesn't exist in the container.

Reply:
Added a condition if the non-sudo fails it will retry with sudo.